### PR TITLE
Make RUBY_RELEASE_DATE specs work across the new year

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: spec/truffle/identity_spec.rb [hunk 1]
```diff
@@ -33,7 +33,8 @@
   end
 
   it "RUBY_RELEASE_DATE is the current year" do
```
Comment (eregon) [inline spec/truffle/identity_spec.rb]: Should the description be changed as well?

**Comment by eregon** [comment]
`RUBY_RELEASE_DATE` is `COMPILE_DATE`, which is hardcoded currently:
https://github.com/graalvm/truffleruby/blob/80e1d9a553a4f5eccd317683ea076af3cc3e0c43/src/main/java/org/truffleruby/RubyLanguage.java#L63

Maybe we should have an annotation processor that generates a file of constants, so the time of `jt/mx build` (`javac` in the end) is recorded (and possibly other information like commit hash). Taking the commit date of the latest commit would even be more accurate and stable.

Not for this PR obviously, but here are some thoughts.
About the spec, should we just check it's a 4-digits number? This spec should pass on a SVM image from 2 years ago, even though it's not likely we'll test that.

File: spec/truffle/identity_spec.rb (hunk 1 above)
Comment (chrisseaton) [reply spec/truffle/identity_spec.rb]: It's just an edge case.

**Comment by chrisseaton** [comment]
I think @nirvdrum was going to try putting the commit hash and build date into the source code somehow, which will fix this properly.

I just realised I didn't want to break CI over the new year.
